### PR TITLE
Create Github action for Nightly install

### LINKED ISSUE #753: Extra dependencies windows installation error
**Description of the error**
Running from a Windows 10 OS, when installing FitBenchmarking via pip from the windows powershell using the first installation command in the installation documentation

    python -m pip install fitbenchmarking[bumps,DFO,minuit,SAS,numdifftools]
 
I encountered several errors and it took some manual intervention to complete the installation. First I encountered an error pertaining to C++ installation:

    error: Microsoft Visual C++ 14.0 is required. Get it with "Build Tools for Visual Studio": 
    https://visualstudio.microsoft.com/downloads/ 

After installing Microsoft Visual Studio 2019, this error disappeared and then I encountered a warning about the *numdifftools* installation (`WARNING: fitbenchmarking 0.1.0 does not provide the extra 'numdifftools' ` and an error relating to the *Minuit* fitting package installation:

    Using legacy 'setup.py install' for iminuit, since package 'wheel' is not installed. 
    Installing collected packages: sasmodels, iminuit, fitbenchmarking 
        Running setup.py install for iminuit ... error
             creating build\temp.win-amd64-3.9\Release\Users\cph47349\AppData\Local\Temp\pip-install-mmf66kj8\iminuit_a383db82cc054e3e98adf0b5a1073e5c\extern\root\math\minuit2 
             error: could not create 'build\temp.win-amd64-3.9\Release\Users\cph47349\AppData\Local\Temp\pip-install-mmf66kj8\iminuit_a383db82cc054e3e98adf0b5a1073e5c\extern\root\math\minuit2': The filename or extension is too 
             long

I installed the *wheel* package and retried but then got a similar error:

        Building wheels for collected packages: iminuit 
          Building wheel for iminuit (setup.py) ... error
        error: could not create 'build\temp.win-amd64-3.9\Release\Users\cph47349\AppData\Local\Temp\pip-install-jtz_kn7u\iminuit_d64f72fbc74042bfa456a820116e40e5\extern\root\math\minuit2': The filename or extension is too long 

**Describe the expected result**
I expected FitBenchmarking and the extra-dependencies to install using pip with no intervention.

**Describe the actual result**
Following the running of `python -m pip install fitbenchmarking[bumps,DFO,minuit,SAS,numdifftools]` I had to install visual studio 2019 for C++ (not mentioned as a requirement in documentation) and due to the error shown in the first section I resorted to using the pip installer for the *Minuit* and *numdifftools*:

      python -m pip install iminuit
      python -m pip install numdifftools

I followed this with a re-run of  `python -m pip install fitbenchmarking[bumps,DFO,minuit,SAS,numdifftools]` thinking it may notify me that all required packages were already installed but it tried to uninstall the existing iminuit package then reinstall it which resulted in the same error `The filename or extension is too long` followed by `Rolling back uninstall of iminuit`.

**Suggested fix**
Provide different advice to windows 10 users whilst this installation issue is fixed, maybe suggest the extra dependency packages are installed as separate commands in pip.

**Additional details**
N/A

### PR DESCRIPTION
**Pull request by robertapplin**
#### Description of Work
This PR adds a job in the github actions to test that fitbenchmarking can be installed as expected using 
```
python -m pip install fitbenchmarking[bumps,DFO,minuit,SAS,numdifftools]
```

It then fixes any problems arising from this new test.

Fixes #753

#### Testing Instructions

1. Make sure that the new job in the github actions passes

Function: Does the change do what it's supposed to?

Tests: Does it pass? Is there adequate coverage for new code?

Style: Is the coding style consistent? Is anything overly confusing?

Documentation: Is there a suitable change to documentation for this change?

### CODE REVIEW

**Comment by robertapplin** [comment]
Confirmed that the nightly build for a release install on windows (`@github-actions
Nightly Build / Release Install Test (Windows)`) will fail for Python 3.9, but not for Python 3.8.

**Comment by robertapplin** [comment]
| First Header  | Second Header |
| ------------- | ------------- |
| Content Cell  | Content Cell  |
| Content Cell  | Content Cell  |

**Comment by tyronerees** [comment]
I think:
- both Windows and Linux
- from pip (I think the release is already covered in other tests, although we can add this later if needed)
- I'd start with 3.9, and we could always add more down the line.

**Comment by robertapplin** [comment]
> I think:
> 
> * both Windows and Linux
> * from pip (I think master is already covered in other tests, although we can add this later if needed)
> * I'd start with 3.9, and we could always add more down the line.

Sounds good. If we merge this before #924 then we should see that the windows py3.9 nightly will fail at midnight (hopefully)

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    runs-on: 'ubuntu-latest'
+    steps:
+    - name: 'Choose Python Version'
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.9'
+    - name: Installing
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install fitbenchmarking[bumps,DFO,gradient_free,minuit,SAS,numdifftools]
```
Comment (tyronerees) [inline .github/workflows/nightly.yml:19]:
```suggestion
        python -m pip install fitbenchmarking[bumps,DFO,minuit,SAS]
```

File: .github/workflows/nightly.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    runs-on: 'windows-latest'
+    steps:
+    - name: 'Choose Python Version'
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.9'
+    - name: Installing
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install fitbenchmarking[bumps,DFO,gradient_free,minuit,SAS,numdifftools]
```
Comment (tyronerees) [inline .github/workflows/nightly.yml:32]:
```suggestion
        python -m pip install fitbenchmarking[bumps,DFO,minuit,SAS]
```

**Review by tyronerees — CHANGES_REQUESTED** [review]
`gradient_free` and `numdifftools` are options in the master branch, but aren't supported in the lastest release (version 0.1.4 -- https://fitbenchmarking.readthedocs.io/en/v0.1.4/users/install_instructions/fitbenchmarking.html)
This looks good once we remove these -- thanks!

**Review by tyronerees — DISMISSED** [review]

**Review by tyronerees — APPROVED** [review]

**Pull request merged**